File: Task-management/task-management/src/main/java/com/taskmanagement/service/UserService.java
package com.taskmanagement.service;

import com.taskmanagement.dto.LoginRequest;
import com.taskmanagement.dto.RegisterRequest;
import com.taskmanagement.dto.UserResponse;
import com.taskmanagement.entity.User;

import java.util.List;

public interface UserService {
    User register(RegisterRequest request);
    User login(LoginRequest request);

    UserResponse getUserByEmail(String email);
    List<UserResponse> getAllUsers();
    UserResponse getUser(Long id);
    List<UserResponse> searchByName(String name);
    List<UserResponse> searchByDepartment(String department);
    UserResponse updateUser(Long id, RegisterRequest request);
    void deleteUser(Long id);
}

File: Task-management/task-management-ui/src/services/userService.js
import API from "./api";

export const getUsers = () => API.get("/users");

export const getUser = (id) => API.get(`/users/${id}`);

export const updateUser = (id, user) =>
    API.put(`/users/${id}`, user);

export const deleteUser = (id) =>
    API.delete(`/users/${id}`);

export const searchUsers = (name) =>
    API.get(`/users/search?name=${name}`);

export const searchDepartment = (department) =>
    API.get(`/users/department?department=${department}`);